data_analysis/model_ad/notebooks/MG-88_Create_harmonized_biomarkers_source_file.rmd

```diff
@@ -0,0 +1,408 @@
+
+# Create Biomarker source file (MG-88)
+
+This notebook generates a single harmonized source file containing the data required to surface Biomarker evidence in the Model-AD Explorer.
+
+Required packages (will automatically be installed if necessary):
+
+- synapser (also requires a Synapse account with an auth token)
+- tidyverse
+
+
+## Library setup
+```{r}
+if (!require("synapser", quietly = TRUE)) {
+  install.packages("synapser", repos = c("http://ran.synapse.org",
+                                         "http://cran.fhcrc.org"))
+}
+
+if (!require("tidyverse", quietly = TRUE)) {
+  install.packages("tidyverse")
+}
+
+library(synapser)
+library(tidyverse)
+```
+
+## Utility functions & variables
+```{r}
+# Harmonized column spec for intermediate per-model files
+keeps <- c('individualID', 'specimenID', 'model', 'type', 'measurement', 'units')
+
+# Vector to store all downloaded synIds for Synapse provenance
+data_sources <- character()
+
+
+# Download a file from Synapse by synId and read it into a dataframe
+download_file <- function(synId, df_name, type) {
+
+  synLogin()
+
+  # Capture all downloaded synIds for provenance
+  data_sources <<- c(data_sources, synId)
+
+  input_dir <- file.path("..", "input")
+  id_parts <- strsplit(synId, '.', fixed = TRUE)
+  id_value <- id_parts[[1]][1]
+  version_value <- id_parts[[1]][2]
+
+  if (is.na(version_value)) {
+    version_value <- NULL
+  }
+
+  cat(paste0("Downloading ", synId, "..."))
+
+
+  file <- synGet(id_value, version = version_value, downloadLocation = input_dir, ifcollision='overwrite.local')
+  path <- file$path
+
+  cat("\nDownload on:  ", date())
+  cat("\n", synId, "\n- Modified on ", file$properties$modifiedOn, "\n- Version ", file$properties$versionNumber)
+  cat("\npath: ", file$path, "\n")
+
+  if (type == 'f') {
+    data <- read_feather(path)
+  } else {
+    if (type == 'json') { data <- fromJSON(path) }
+    if (type == 'csv' || type == 'table') { data <- read.csv(path) }
+    if (type == 'tsv') { data <- read.csv(path, sep = "\t") }
+    if (type == 'rda') { data <- load(path)}
+    if (type == 'rds') {data <- readRDS(path)}
+    df_name <- as.data.frame(data)
+  }
+}
+
+# Join a data file to a pair of metadata files to get required age, sex, ageDeath, and genotype values
+join_to_metadata <- function(df, individual_metadata, biospecimen_metadata) {
+
+  individual_metadata <- subset(individual_metadata, select=c(individualID, sex, ageDeath, genotype))
+  biospecimen_metadata <- subset(biospecimen_metadata, select=c(individualID, specimenID, tissue))
+  metadata <- merge(individual_metadata, biospecimen_metadata, by = 'individualID')
+
+  #df_out <- merge(df, metadata) # MG-12 - drop rows that are missing from biospecimen metadata
+  df_out <- merge(df, metadata, fixed=TRUE)
```

Comment:
I can't find any documentation that "merge" has a "fixed" argument -- what does this do?

Reply:
I have no idea why I had that there, I've removed it now.